File: Task1/ObjectiveC/Cocktail.m
#import "Cocktail.h"

@interface Cocktail ()
@property (nonatomic, nonnull, strong) NSString *name;
@property (nonatomic, assign) float baseStrength;
@property (nonatomic, assign) int baseVolume;
@property (nonatomic, assign) int nonAlcoholicBeveragesVolume;

@end

@implementation Cocktail

- (instancetype)initWithName:(NSString *)name baseStrength:(float)baseStrength baseVolume:(int)baseVolume nonAlcoholicBeveragesVolume:(int)nonAlcoholicBeveragesVolume {
    if (self = [super init]) {
        _name = name;
        _baseStrength = baseStrength;
        _baseVolume = baseVolume;
        _nonAlcoholicBeveragesVolume = nonAlcoholicBeveragesVolume;
    }

    return self;
}

@end

File: Cocktail/Swift/Cocktail.swift
import Foundation

struct Cocktail {
    private let name: String
    private let baseStrength: Float
    private let baseVolume: Int
    private let nonAlcoholicBeveragesVolume: Int
    
    init(name: String, baseStrength: Float, baseVolume: Int, nonAlcoholicBeveragesVolume: Int) {
        self.name = name
        self.baseStrength = baseStrength
        self.baseVolume = baseVolume
        self.nonAlcoholicBeveragesVolume = nonAlcoholicBeveragesVolume
    }
}
